packages/next/src/build/turborepo-access-trace/env.ts

```diff
@@ -0,0 +1,32 @@
+import type { EnvVars, RestoreOriginalFunction } from './types'
+
+/**
+ * Proxy the environment to track environment variables keys that
+ * are accessed during the build.
+ *
+ * @param envVars A set to track environment variable keys that are accessed.
+ * @returns A function that restores the original environment.
+ */
+export function envProxy(envVars: EnvVars): RestoreOriginalFunction {
+  const newEnv = new Proxy(process.env, {
+    get: (target, key) => {
+      envVars.add(key)
+      return target[key as string]
+    },
+    getOwnPropertyDescriptor(_target, _key) {
+      return { configurable: true, enumerable: true }
+    },
+    set: (target, key, value) => {
+      target[key as string] = value
+      return true
+    },
```

Comment:
As this isn't overriding any specific behaviours, you can exclude these traps.

Reply:
Since I'm replacing the entire `process.env` with this proxy I do need to override `set` otherwise it won't be accessible (and will error if used). process.env behaves differently when replaces with a Proxy and doesn't seem to Reflect by default. 

However I did update `set` to just `Reflect` so it's just pass through.